src/database/DataTypes/StocktakeItem.js

```diff
@@ -243,6 +243,76 @@ export class StocktakeItem extends Realm.Object {
     const itemBatch = createRecord(database, 'ItemBatch', this.item, batchString);
     createRecord(database, 'StocktakeBatch', this, itemBatch, true);
   }
+
+  /**
+   * Applies the given Options object to all stocktake batches assosciated to
+   * this stocktake item, if there is a difference between countedTotalQuantity
+   * and snapshotTotalQuantity.
+   * @param {Realm}   database
+   * @param {Options} option
+   */
+  applyReasonToBatches(database, option) {
+    this.batches.forEach(batch => {
+      if (batch.countedTotalQuantity !== batch.snapshotTotalQuantity) {
+        database.write(() => {
+          database.update('StocktakeBatch', { id: batch.id, option });
+        });
+      }
+    });
+  }
+
+  /**
+   * Removes all Options objects from the stocktake batches assosciated
+   * with this stocktake item.
+   * @param {Realm} database
+   */
+  removeReason(database) {
+    this.batches.forEach(batch => {
+      if (batch.option) {
+        database.write(() => {
+          database.update('StocktakeBatch', { ...batch, option: null });
+        });
+      }
+    });
+  }
+
+  /**
+   * Finds the mode option within this stocktake items batches
+   * @return {string} The title of the reason with the highest frequency
+   */
+  get mostUsedReason() {
+    if (!this.batches) return false;
+    if (!this.batches[0]) return false;
```

Comment:
I think my unfamiliarity with realm. Can `this.batches === undefined/null` ? I guess even then `this.batches && this.batches.length === 0)` is nicer

Reply:
> Can `this.batches === undefined/null`

`!this.batches` already catches that, it does not however catch an empty array/list which `this.batches.length === 0` does. Interestingly you could do `!this.batches.length`, as 0 is a falsy value. But I think being consistent with the code base is probably better:
![image](https://user-images.githubusercontent.com/7684221/55522464-97726e00-56e1-11e9-82c7-5bca2fbd45de.png)

All that said, `!this.batches.[0]` might not behave the way you want in other contexts (i.e. I don't think this can happen with realm): `![null, 2, 3][0] // true`

Reply:
done 6b01afbd4ac070153b2f98cee60061ac8416ee38
Also tried to make `get mostUsedReason` more readable